### ISSUE TITLE
JIT: consider inlining synchronized methods

### ISSUE THREAD

**Issue by AndyAyersMS** [dotnet/coreclr#11189]
Synchronized methods aren't inlined currently, even with the aggressive inlining attribute. This may end up causing performance surprises.

**Comment by mikedn** [comment]
Synchronized as in `[MethodImpl(MethodImplOptions.Synchronized)]`? I'd be surprised if those cause performance surprises, I haven't seen one in ages.

**Comment by AndyAyersMS** [comment]
@cmckinsey may want to add in some examples he's seen...

**Comment by BruceForstall** [comment]
I don't think we can do it on x86, due to synchronized methods being reported in GC info. Non-x86 we need to inject try/finally or try/fault; doing that on an inlined function seems tricky. We don't inline any function with EH currently, do we?
